onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -25,23 +26,29 @@ inline AllocatorPtr& GetAllocator() {
 class SessionObjectInitializer {
  public:
   typedef const SessionOptions& Arg1;
-  typedef logging::LoggingManager* Arg2;
+  // typedef logging::LoggingManager* Arg2;
+  static const std::string default_logger_id;
   operator Arg1() {
     return GetDefaultCPUSessionOptions();
   }
 
-  operator Arg2() {
-    static std::string default_logger_id{"Default"};
-    static LoggingManager default_logging_manager{std::unique_ptr<ISink>{new CErrSink{}},
-                                                  Severity::kWARNING, false, LoggingManager::InstanceType::Default,
-                                                  &default_logger_id};
-    return &default_logging_manager;
-  }
+  // operator Arg2() {
+  //   static LoggingManager default_logging_manager{std::unique_ptr<ISink>{new CErrSink{}},
+  //                                                 Severity::kWARNING, false, LoggingManager::InstanceType::Default,
+  //                                                 &default_logger_id};
+  //   return &default_logging_manager;
+  // }
 
   static SessionObjectInitializer Get() {
     return SessionObjectInitializer();
   }
 };
+
+// static variable used to create inference session and training session.
+static std::unique_ptr<Environment> session_env;
```

Comment:
being a static variable in .h each compilation unit (.o file) will have its own session_env. 
put session_env in onnxruntime_pybind_state.cc in front of initialize_env so that session_env is shared between ortraining_pybind_state and onnxruntime_pubind_state.

Reply:
@pranavsharma Please take a look. The session_env can't be a static variable if it contains a thread pool. In this special case, it may looks ok, because it is not using the new feature. But eventually, we need to fix it.  At some time, people will ask why they can't use the share thread pool in python API.

Reply:
Can we please avoid this?

Reply:
@pranavsharma, not sure what you mean.. This is copied from onnxruntime_pybind_stage, and it is defined as a static variable there.

Reply:
Yeah, it was in the .cc file before.

Reply:
Sure, move it back in .cc file is fine to me.

Reply:
I think move it back to .cc won't work. If session_env is never declared in .h file, how would ortraining_pybind_state.cc know about it? `extern` keyword won't work with static.

Given that static is only local to translation unit, if we want to make it sharable across onnx and ort, we need to remote the static from it, which would be a design change. I am lacking of context on making the right call, will defer this to @pranavsharma and @liqunfu .